### PR TITLE
[openexr] fix iOS build, set bundle suffix for iOS

### PR DESCRIPTION
**Pull request by russelltg**
- [openexr] fix iOS build, set bundle suffix for iOS
- ./vcpkg x-add-version --all

This switches to using vcpkg_copy_tools for openexr, which works in iOS as it uses the bundle extension. 

**Describe the pull request**

- #### What does your PR fix?
  Fixes openexr package on iOS

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  arm64-ios, arm-ios

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by dg0yt** [comment]
Maybe you wait with openexr. There is a PR for updating to 2.5.8 (https://github.com/microsoft/vcpkg/pull/26503). 
And the real challenge is finding a vcpkg way to transition to version 3.

**Comment by russelltg** [comment]
I see, good call. 

I still think my change to [scripts/cmake/vcpkg_common_definitions.cmake](https://github.com/microsoft/vcpkg/pull/26523/files#diff-de084a08a84c18cb864ebcd1cd57d6560d31275c04ece2ddf7804e15ba8ae769) is valid, should I open a new PR?

**Comment by dg0yt** [comment]
The changes to `scripts/cmake/...` cause a lot of packages to rebuild. It may make sense to bundle the iOS changes but YMMV. Reviewers, contributors and users may have different perspectives and demands.
FTR there is a set of scripts changes for Android, but we didn't manage yet to get them through reviews...

File: ports/openexr/portfile.cmake [hunk 1]
```diff
@@ -15,37 +15,22 @@ vcpkg_from_github(
...
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exrmakepreview${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exrmakepreview${VCPKG_HOST_EXECUTABLE_SUFFIX})
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exrmaketiled${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exrmaketiled${VCPKG_HOST_EXECUTABLE_SUFFIX})
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exrmultipart${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exrmultipart${VCPKG_HOST_EXECUTABLE_SUFFIX})
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exrmultiview${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exrmultiview${VCPKG_HOST_EXECUTABLE_SUFFIX})
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exrstdattr${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exrstdattr${VCPKG_HOST_EXECUTABLE_SUFFIX})
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/exr2aces${VCPKG_HOST_EXECUTABLE_SUFFIX} ${CURRENT_PACKAGES_DIR}/tools/openexr/exr2aces${VCPKG_HOST_EXECUTABLE_SUFFIX})
+vcpkg_copy_tools(
+  TOOL_NAMES exrenvmap exrheader exrmakepreview exrmaketiled exrmultipart exrmultiview exrstdattr exr2aces
+  AUTO_CLEAN
+)
```
Comment (dg0yt) [inline ports/openexr/portfile.cmake:32]: Better test changes like this separately from changes triggering world-rebuild.

**Comment by russelltg** [comment]
Makes sense. Should I bundle my scripts/cmake changes from this and #26520 into a separate PR then, and mark #26520 as a draft until that change is merged?

**Comment by JackBoosY** [comment]
Please get failure logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=77055&view=artifacts&pathAsName=false&type=publishedArtifacts).

**Comment by russelltg** [comment]
Was going through the logs with the intent of opening a new PR with just the bundle suffix changes, and they seemed quite unrelated:
```
Half-2_5.lib(half.cpp.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in spin_rw_test.cpp.obj
Half-2_5.lib(half.cpp.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MTd_StaticDebug' in spin_rw_test.cpp.obj
```

Suppose I'll just go for it and see

**Comment by russelltg** [comment]
I ended up splitting this PR into #26612 and #26641

**Pull request closed without merging**